File: Java/Aplicatii client-server/labJavaClientServerProtoBuf/Networking/src/main/java/ProtocolBuffers/ProtoUtils.java
package ProtocolBuffers;

import javafx.util.Pair;
import model.Cursa;
import model.Oficiu;
import model.Participant;
import model.ParticipantCursa;

public class ProtoUtils {


    //REQUESTS
    public static Protocol.Request createLoginRequest(Oficiu oficiu){
        Protocol.OficiuDTO oficiuDTO = Protocol.OficiuDTO.newBuilder()
                .setIdOficiu(oficiu.getId())
                .setUsername(oficiu.getUsername())
                .setPassword(oficiu.getPassword())
                .build();
        Protocol.Request request = Protocol.Request.newBuilder()
                .setType(Protocol.Request.Type.Login)
                .setOficiuDTO(oficiuDTO)
                .build();
        return request;
    }

    public static Protocol.Request createGetOficiuByUserPassRequest(String username, String password){
        Protocol.Request request = Protocol.Request.newBuilder()
                .setType(Protocol.Request.Type.GetOficiuByUserPass)
                .setUsername(username)
                .setPassword(password)
                .build();
        return request;
    }

    public static Protocol.Request createLogoutRequest(Oficiu oficiu){
        Protocol.OficiuDTO oficiuDTO = Protocol.OficiuDTO.newBuilder()
                .setIdOficiu(oficiu.getId())
                .setUsername(oficiu.getUsername())
                .setPassword(oficiu.getPassword())
                .build();
        Protocol.Request request = Protocol.Request.newBuilder()
                .setType(Protocol.Request.Type.Logout)
                .setOficiuDTO(oficiuDTO)
                .build();
        return request;
    }

    public static Protocol.Request createGetAllParticipantsRequest(){
        Protocol.Request request = Protocol.Request.newBuilder()
                .setType(Protocol.Request.Type.GetAllParticipants)
                .build();
        return request;
    }

    public static Protocol.Request createGetAllCursaRequest(){
        Protocol.Request request = Protocol.Request.newBuilder()
                .setType(Protocol.Request.Type.GetAllCursa)
                .build();
        return request;
    }

    public static Protocol.Request createGetAllParticipantsByTeamRequest(String echipa){
        Protocol.Request request = Protocol.Request.newBuilder()
                .setType(Protocol.Request.Type.GetAllParticipantsByTeam)
                .setEchipa(echipa)
                .build();
        return request;
    }

    public static Protocol.Request createGetEchipeRequest(){
        Protocol.Request request = Protocol.Request.newBuilder()
                .setType(Protocol.Request.Type.GetEchipe)
                .build();
        return request;
    }

    public static Protocol.Request createGetParticipantRequest(String nume, String echipa, Integer capMotor){
        Protocol.Request request = Protocol.Request.newBuilder()
                .setType(Protocol.Request.Type.GetParticipant)
                .setNume(nume)
                .setEchipa(echipa)
                .setCapMotor(capMotor)
                .build();
        return request;
    }

    public static Protocol.Request createGetCursaRequest(String idCursa){
        Protocol.Request request = Protocol.Request.newBuilder()
                .setType(Protocol.Request.Type.GetCursa)
                .setIdCursa(idCursa)
                .build();
        return request;
    }

    public static Protocol.Request createAddParticipantCursaRequest(ParticipantCursa participantCursa){
        Protocol.ParticipantCursa participantCursaDTO = Protocol.ParticipantCursa.newBuilder()
                .setIdParticipant(participantCursa.getIdParticipant())
                .setIdCursa(participantCursa.getIdCursa())
                .build();
        Protocol.Request request = Protocol.Request.newBuilder()
                .setType(Protocol.Request.Type.AddParticipantCursa)
                .setParticipantCursa(participantCursaDTO)
                .build();
        return request;
    }

    public static Protocol.Request createUpdateCursa(String idCursa, Cursa cursa){
        Protocol.CursaDTO cursaDTO = Protocol.CursaDTO.newBuilder()
                .setIdCursa(cursa.getId())
                .setNrPersoane(cursa.getNrPersoane())
                .setCapMotor(cursa.getCapacitateMotor())
                .build();
        Protocol.Request request = Protocol.Request.newBuilder()
                .setType(Protocol.Request.Type.UpdateCursa)
                .setIdCursa(idCursa)
                .setCursaDTO(cursaDTO)
                .build();
        return request;
    }


    //RESPONSES
    public static Protocol.Response createOkReponse(){
        Protocol.Response response = Protocol.Response.newBuilder()
                .setType(Protocol.Response.Type.Ok)
                .build();
        return response;
    }

    public static Protocol.Response createErrorResponse(String eroare){
        Protocol.Response response = Protocol.Response.newBuilder()
                .setType(Protocol.Response.Type.Error)
                .setEroare(eroare)
                .build();
        return response;
    }

    public static Protocol.Response createGetOficiuByUserPassResponse(Oficiu oficiu){
        Protocol.OficiuDTO oficiuDTO = Protocol.OficiuDTO.newBuilder()
                .setIdOficiu(oficiu.getId())
                .setUsername(oficiu.getUsername())
                .setPassword(oficiu.getPassword())
                .build();
        Protocol.Response response = Protocol.Response.newBuilder()
                .setType(Protocol.Response.Type.GetOficiuByUserPass)
                .setOficiuDTO(oficiuDTO)
                .build();
        return response;
    }

    public static Protocol.Response createGetAllCursaResponse(Cursa[] curse){
        Protocol.Response.Builder response = Protocol.Response.newBuilder()
                .setType(Protocol.Response.Type.GetAllCursa);

        for (Cursa cursa : curse){
            Protocol.CursaDTO cursaDTO = Protocol.CursaDTO.newBuilder()
                    .setIdCursa(cursa.getId())
                    .setNrPersoane(cursa.getNrPersoane())
                    .setCapMotor(cursa.getCapacitateMotor())
                    .build();
            response.addCursaDTO(cursaDTO);
        }
        return response.build();
    }

    public static Protocol.Response createGetAllParticipantsResponse(Participant[] participants){
        Protocol.Response.Builder response = Protocol.Response.newBuilder()
                .setType(Protocol.Response.Type.GetAllParticipants);
        for (Participant participant : participants){
            Protocol.Participant participantDTO = Protocol.Participant.newBuilder()
                    .setNume(participant.getNume())
                    .setEchipa(participant.getEchipa())
                    .setCapMotor(participant.getCapacitateMotor())
                    .build();
            response.addParticipant(participantDTO);
        }
        return response.build();
    }

    public static Protocol.Response createGetEchipeResponse(String echipa){
        Protocol.Response response = Protocol.Response.newBuilder()
                .setType(Protocol.Response.Type.GetEchipe)
                .setEchipe(echipa)
                .build();
        return response;
    }

    public static Protocol.Response createGetParticipantResponse(Participant participant){
        Protocol.ParticipantDTO participantDTO = Protocol.ParticipantDTO.newBuilder()
                .setIdParticipant(participant.getId())
                .setNume(participant.getNume())
                .setEchipa(participant.getEchipa())
                .setCapMotor(participant.getCapacitateMotor())
                .build();
        Protocol.Response response = Protocol.Response.newBuilder()
                .setType(Protocol.Response.Type.GetParticipant)
                .setParticipantDTO(participantDTO)
                .build();
        return response;
    }

    public static Protocol.Response createGetCursaResponse(Cursa cursa){
        Protocol.Cursa cursaDTO = Protocol.Cursa.newBuilder()
                .setNrPersoane(cursa.getNrPersoane())
                .setCapMotor(cursa.getCapacitateMotor())
                .build();
        Protocol.Response response = Protocol.Response.newBuilder()
                .setType(Protocol.Response.Type.GetCursa)
                .setCursa(cursaDTO)
                .build();
        return response;
    }

    public static Protocol.Response createNewParticipantCursaAddedResponse(){
        Protocol.Response response = Protocol.Response.newBuilder()
                .setType(Protocol.Response.Type.NewParticipantCursaAdded)
                .build();
        return response;
    }


    //GET from REQUESTS
    public static Oficiu getOficiu(Protocol.Request request){
        Oficiu oficiu = new Oficiu("", "");
        oficiu.setId(request.getOficiuDTO().getIdOficiu());
        oficiu.setUsername(request.getOficiuDTO().getUsername());
        oficiu.setPassword(request.getOficiuDTO().getPassword());
        return oficiu;
    }

    public static String getUsername(Protocol.Request request){
        return request.getUsername();
    }

    public static String getPassword(Protocol.Request request){
        return request.getPassword();
    }

    public static String getEchipa(Protocol.Request request){
        return request.getEchipa();
    }

    public static String getNume(Protocol.Request request){
        return request.getNume();
    }

    public static Integer getCapMotor(Protocol.Request request){
        return request.getCapMotor();
    }

    public static String getIdCursa(Protocol.Request request){
        return request.getIdCursa();
    }

    public static ParticipantCursa getParticipantCursa(Protocol.Request request){
        ParticipantCursa participantCursa = new ParticipantCursa("", "");
        Pair<String, String> id = new Pair<>(request.getParticipantCursa().getIdParticipant(), request.getParticipantCursa().getIdCursa());
        participantCursa.setId(id);
        participantCursa.setIdParticipant(request.getParticipantCursa().getIdParticipant());
        participantCursa.setIdCursa(request.getParticipantCursa().getIdCursa());
        return participantCursa;
    }

    public static Cursa getCursa(Protocol.Request request){
        Cursa cursa = new Cursa(0, 0);
        cursa.setId(request.getCursaDTO().getIdCursa());
        cursa.setNrPersoane(request.getCursaDTO().getNrPersoane());
        cursa.setCapacitateMotor(request.getCursaDTO().getCapMotor());
        return cursa;
    }


    //GET from RESPONSES
    public static String getEroare(Protocol.Response response){
        return response.getEroare();
    }

    public static Oficiu getOficiu(Protocol.Response response){
        Oficiu oficiu = new Oficiu("", "");
        oficiu.setId(response.getOficiuDTO().getIdOficiu());
        oficiu.setUsername(response.getOficiuDTO().getUsername());
        oficiu.setPassword(response.getOficiuDTO().getPassword());
        return oficiu;
    }

    public static Cursa[] getCurse(Protocol.Response response){
        Cursa[] curse = new Cursa[response.getCursaDTOCount()];
        for(int i = 0; i < response.getCursaDTOCount(); i++){
            Protocol.CursaDTO cursaDTO = response.getCursaDTO(i);
            Cursa cursa = new Cursa(0, 0);
            cursa.setId(cursaDTO.getIdCursa());
            cursa.setNrPersoane(cursaDTO.getNrPersoane());
            cursa.setCapacitateMotor(cursaDTO.getCapMotor());
            curse[i] = cursa;
        }
        return curse;
    }

    public static Participant[] getParticipants(Protocol.Response response){
        Participant[] participants = new Participant[response.getParticipantCount()];
        for(int i = 0; i < response.getParticipantCount(); i++){
            Protocol.Participant participantDTO = response.getParticipant(i);
            Participant participant = new Participant("", "", 0);
            participant.setNume(participantDTO.getNume());
            participant.setEchipa(participantDTO.getEchipa());
            participant.setCapacitateMotor(participantDTO.getCapMotor());
            participants[i] = participant;
        }
        return participants;
    }

    public static String getEchipe(Protocol.Response response){
        return response.getEchipe();
    }

    public static Participant getParticipant(Protocol.Response response){
        Participant participant = new Participant("", "", 0);
        //System.out.println("response ul primit in ProtoUtils este: " + response.getCursa());
        System.out.println("participantulDTO primit in ProtoUtils este: " + response.getParticipantDTO());
        Protocol.ParticipantDTO participantDTO = response.getParticipantDTO();
        participant.setId(participantDTO.getIdParticipant());
        participant.setNume(participantDTO.getNume());
        participant.setEchipa(participantDTO.getEchipa());
        participant.setCapacitateMotor(participantDTO.getCapMotor());
        return participant;
    }

    public static Cursa getCursa(Protocol.Response response){
        Cursa cursa = new Cursa(0, 0);
        System.out.println("cursa primita in ProtoUtils este: " + response.getCursa());
        Protocol.Cursa cursaDTO = response.getCursa();
        cursa.setNrPersoane(cursaDTO.getNrPersoane());
        cursa.setCapacitateMotor(cursaDTO.getCapMotor());
        return cursa;
    }


    //
}

File: C#/Aplicatii client-server/labClientServerCSharpProtoBuf/ClientServer/Protobuff/ProtoUtils.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using lab3.domain;
using Protobuff;

namespace Protobuff
{
    static class ProtoUtils
    {
        //REQUESTS
        //public static Request createLoginRequest(Oficiu oficiu)
        //{
        //    OficiuDTO oficiuDTO = new OficiuDTO
        //    {
        //        IdOficiu = oficiu.Id,
        //        Username = oficiu.Username,
        //        Password = oficiu.Password
        //    };
        //    Request request = new Request
        //    {
        //        Type = Request.Types.Type.Login,
        //        OficiuDTO = oficiuDTO
        //    };
        //    return request;
        //}

        //public static Request createLogoutRequest(Oficiu oficiu)
        //{
        //    OficiuDTO oficiuDTO = new OficiuDTO
        //    {
        //        IdOficiu = oficiu.Id,
        //        Username = oficiu.Username,
        //        Password = oficiu.Password
        //    };
        //    Request request = new Request
        //    {
        //        Type = Request.Types.Type.Logout,
        //        OficiuDTO = oficiuDTO
        //    };
        //    return request;
        //}

        //public static Request createFindOficiuByUserPass(string username, string password)
        //{
        //    Request request = new Request
        //    {
        //        Type = Request.Types.Type.FindOficiuByUserPass,
        //        Username = username,
        //        Password = password
        //    };
        //    return request;
        //}

        //public static Request createFindAllParticipant()
        //{
        //    Request request = new Request
        //    {
        //        Type = Request.Types.Type.FindAllParticipant
        //    };
        //    return request;
        //}

        //public static Request createFindAllCursa()
        //{
        //    Request request = new Request
        //    {
        //        Type = Request.Types.Type.FindAllCursa
        //    };
        //    return request;
        //}

        //public static Request createFindAllParticipantEchipa(string echipa)
        //{
        //    Request request = new Request
        //    {
        //        Type = Request.Types.Type.FindAllParticipantEchipa,
        //        Echipa = echipa
        //    };
        //    return request;
        //}

        //public static Request createFindParticipant(string nume, string echipa, int capMotor)
        //{
        //    Request request = new Request
        //    {
        //        Type = Request.Types.Type.FindParticipant,
        //        Nume = nume,
        //        Echipa = echipa,
        //        CapMotor = capMotor
        //    };
        //    return request;
        //}

        //public static Request createFindCursa(string idCursa)
        //{
        //    Request request = new Request
        //    {
        //        Type = Request.Types.Type.FindCursa,
        //        IdCursa = idCursa
        //    };
        //    return request;
        //}

        //public static Request createSaveParticipantCursa(ParticipantCursa participantCursa)
        //{
        //    ParticipantCursaDTO participantCursaDTO = new ParticipantCursaDTO
        //    {
        //        IdParticipant = participantCursa.IdParticipant,
        //        IdCursa = participantCursa.IdCursa
        //    };
        //    Request request = new Request
        //    {
        //        Type = Request.Types.Type.SaveParticipantCursa,
        //        ParticipantCursaDTO = participantCursaDTO
        //    };
        //    return request;
        //}

        //public static Request createDeleteCursa(string idCursa)
        //{
        //    Request request = new Request
        //    {
        //        Type = Request.Types.Type.DeleteCursa,
        //        IdCursa = idCursa
        //    };
        //    return request;
        //}

        //public static Request createSaveCursa(Cursa cursa)
        //{
        //    CursaDTO cursaDTO = new CursaDTO
        //    {
        //        IdCursa = cursa.Id,
        //        NrPersoane = cursa.NrPersoane,
        //        CapMotor = cursa.CapacitateMotor
        //    };
        //    Request request = new Request
        //    {
        //        Type = Request.Types.Type.SaveCursa,
        //        CursaDTO = cursaDTO
        //    };
        //    return request;
        //}

        //public static Request createNotify(Cursa[] curse)
        //{
        //    Request request = new Request
        //    {
        //        Type = Request.Types.Type.Notify,
        //    };
        //    foreach (Cursa cursa in curse)
        //    {
        //        CursaDTO cursaDTO = new CursaDTO
        //        {
        //            IdCursa = cursa.Id,
        //            NrPersoane = cursa.NrPersoane,
        //            CapMotor = cursa.CapacitateMotor
        //        };
        //        request.CurseDTO.Add(cursaDTO);
        //    }
        //    return request;
        //}

        //RESPONSES
        public static Response createOkResponse()
        {
            Response response = new Response
            {
                Type = Response.Types.Type.Ok
            };
            return response;
        }

        public static Response createErrorResponse(string eroare)
        {
            Response response = new Response
            {
                Type = Response.Types.Type.Error,
                Eroare = eroare
            };
            return response;
        }

        public static Response createGetOficiuByUserPassResponse(Oficiu oficiu)
        {
            OficiuDTO oficiuDTO = new OficiuDTO
            {
                IdOficiu = oficiu.Id,
                Username = oficiu.Username,
                Password = oficiu.Password
            };
            Response response = new Response
            {
                Type = Response.Types.Type.GetOficiuByUserPass,
                OficiuDTO = oficiuDTO
            };
            return response;
        }

        public static Response createGetAllParticipantResponse(lab3.domain.Participant[] participants)
        {
            Response response = new Response
            {
                Type = Response.Types.Type.GetAllParticipants,
            };
            foreach (lab3.domain.Participant participant in participants)
            {
                Participant participantDTO = new Participant
                {
                    Nume = participant.Nume,
                    Echipa = participant.Echipa,
                    CapMotor = participant.CapacitateMotor
                };
                response.Participanti.Add(participantDTO);
            }
            return response;
        }

        public static Response createGetAllCursaResponse(lab3.domain.Cursa[] curse)
        {
            Response response = new Response
            {
                Type = Response.Types.Type.GetAllCursa,
            };
            foreach (lab3.domain.Cursa cursa in curse)
            {
                CursaDTO cursaDTO = new CursaDTO
                {
                    IdCursa = cursa.Id,
                    NrPersoane = cursa.NrPersoane,
                    CapMotor = cursa.CapacitateMotor
                };
                response.CurseDTO.Add(cursaDTO);
            }
            return response;
        }

        public static Response createGetEchipeResponse(string echipe)
        {
            Response response = new Response
            {
                Type = Response.Types.Type.GetEchipe,
                Echipe = echipe
            };
            return response;
        }

        public static Response createGetParticipantResponse(lab3.domain.Participant participant)
        {
            ParticipantDTO participantDTO = new ParticipantDTO
            {
                IdParticipant = participant.Id,
                Nume = participant.Nume,
                Echipa = participant.Echipa,
                CapMotor = participant.CapacitateMotor
            };
            Response response = new Response
            {
                Type = Response.Types.Type.GetParticipant,
                ParticipantDTO = participantDTO
            };
            return response;
        }

        public static Response createGetCursaResponse(lab3.domain.Cursa cursa)
        {
            Cursa cursaDTO = new Cursa
            {
                NrPersoane = cursa.NrPersoane,
                CapMotor = cursa.CapacitateMotor
            };
            Response response = new Response
            {
                Type = Response.Types.Type.GetCursa,
                Cursa = cursaDTO
            };
            return response;
        }

        public static Response createNewParticipantCursaAddedResponse()
        {
            Response response = new Response
            {
                Type = Response.Types.Type.NewParticipantCursaAdded,
            };
            return response;
        }

        //GET from REQUESTS
        public static Oficiu getOficiu(Request request)
        {
            Oficiu oficiu = new Oficiu(request.OficiuDTO.Username, request.OficiuDTO.Password)
            {
                Id = request.OficiuDTO.IdOficiu
            };
            return oficiu;
        }

        public static string getUsername(Request request)
        {
            return request.Username;
        }

        public static string getPassword(Request request)
        {
            return request.Password;
        }

        public static string getEchipa(Request request)
        {
            return request.Echipa;
        }

        public static string getNume(Request request)
        {
            return request.Nume;
        }

        public static int getCapMotor(Request request)
        {
            return request.CapMotor;
        }

        public static string getIdCursa(Request request)
        {
            return request.IdCursa;
        }

        public static lab3.domain.ParticipantCursa getParticipantCursa(Request request)
        {
            lab3.domain.ParticipantCursa participantCursa = new lab3.domain.ParticipantCursa(request.ParticipantCursa.IdParticipant, request.ParticipantCursa.IdCursa);
            participantCursa.Id = new Tuple<string, string>(request.ParticipantCursa.IdParticipant, request.ParticipantCursa.IdCursa);
            return participantCursa;
        }

        public static lab3.domain.Cursa getCursa(Request request)
        {
            lab3.domain.Cursa cursa = new lab3.domain.Cursa(request.CursaDTO.NrPersoane, request.CursaDTO.CapMotor)
            {
                Id = request.CursaDTO.IdCursa
            };
            return cursa;
        }

        //GET from RESPONSES
        //public static string getEroare(Response response)
        //{
        //    return response.Eroare;
        //}

        //public static Oficiu getOficiu(Response response)
        //{
        //    Oficiu oficiu = new Oficiu(response.OficiuDTO.Username, response.OficiuDTO.Password)
        //    {
        //        Id = response.OficiuDTO.IdOficiu
        //    };
        //    return oficiu;
        //}

        //public static string getEchipe(Response response)
        //{
        //    return response.Echipe;
        //}

        //public static lab3.domain.Participant getParticipant(Response response)
        //{
        //    lab3.domain.Participant participant = new lab3.domain.Participant(response.ParticipantDTO.Nume, response.ParticipantDTO.Echipa, response.ParticipantDTO.CapMotor)
        //    {
        //        Id = response.ParticipantDTO.IdParticipant
        //    };
        //    return participant;
        //}

        //public static lab3.domain.Cursa getCursa(Response response)
        //{
        //    lab3.domain.Cursa cursa = new lab3.domain.Cursa(response.Cursa.NrPersoane, response.Cursa.CapMotor);
        //    return cursa;
        //}

        //public static lab3.domain.Participant[] getParticipants(Response response)
        //{
        //    lab3.domain.Participant[] participants = new lab3.domain.Participant[response.Participant.Count];
        //    for(int i = 0; i < response.Participant.Count; i++)
        //    {
        //        lab3.domain.Participant participant = new lab3.domain.Participant(response.Participant[i].Nume, response.Participant[i].Echipa, response.Participant[i].CapMotor);
        //        participants[i] = participant;
        //    }
        //    return participants;
        //}

        //public static lab3.domain.Cursa[] getCurse(Response response)
        //{
        //    lab3.domain.Cursa[] curse = new lab3.domain.Cursa[response.CurseDTO.Count];
        //    for (int i = 0; i < response.CurseDTO.Count; i++)
        //    {
        //        lab3.domain.Cursa cursa = new lab3.domain.Cursa(response.CurseDTO[i].NrPersoane, response.CurseDTO[i].CapMotor)
        //        {
        //            Id = response.CurseDTO[i].IdCursa
        //        };
        //        curse[i] = cursa;
        //    }
        //    return curse;
        //}
    }
}
